include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -6590,6 +6628,99 @@ struct OrtApi {
    * \since Version 1.24
    */
   ORT_API_T(bool, TensorTypeAndShape_HasShape, _In_ const OrtTensorTypeAndShapeInfo* info);
+
+  /** \brief Setup Graphics Interopcontext for an execution provider device.
+   *
+   * This function enables D3D12/Vulkan interoperability by creating a CUDA context
+   * associated with a graphics API command queue/device. Once setup, any OrtSyncStream
+   * created for this ep_device via CreateSyncStreamForEpDevice will be created on the
+   * CIG context, enabling efficient GPU-side synchronization between graphics APIs and CUDA.
+   *
+   * This must be called BEFORE CreateSyncStreamForEpDevice for the same ep_device if you want
+   * streams to be created on the CIG context.
+   *
+   * \param[in] ep_device The OrtEpDevice to setup CIG context for.
+   * \param[in] graphicsInteropParams Pointer to struct containing D3D12 command queue or Vulkan device info.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.24
+   */
+  ORT_API2_STATUS(SetupGraphicsInteropContextForEpDevice, _In_ const OrtEpDevice* ep_device,
+                  _In_ const struct GraphicsInteropParams* graphicsInteropParams);
+
+  /**
+   * \brief Get an external GPU fence/semaphore suitable for graphics interop.
+   *
+   * This function enables interoperability between ONNX Runtime and graphics APIs by obtaining a GPU external synchronization
+   * primitive (such as a semaphore or fence handle) that may be shared with external systems like D3D12 or Vulkan.
+   *
+   * This allows integration between ONNX Runtime GPU computation and external graphics or compute pipelines for scenarios such
+   * as synchronizing operations between ONNX Runtime's inference execution and mechanisms external to ONNX Runtime.
+   *
+   * The exact semantics and type of the synchronization primitive returned depend on the executing provider and the parameters provided.
+   *
+   * \param[in] session An OrtSession instance whose compute graph and providers participate in the graphics interop.
+   * \param[in] graphicsInteropParams Pointer to a struct defining which graphics API interop is desired, providing descriptor handles, device pointers, etc.
+   * \param[out] extSemFence Pointer to a variable that will receive the external semaphore/fence handle. The handle is suitable for passing to external graphics APIs.
+   *
+   * \retval ORT_OK On success.
+   * \retval ORT_NOT_IMPLEMENTED If none of the active providers support graphics-fence interop.
+   * \retval ORT_FAIL or provider-specific error if the operation fails for another reason.
+   *
+   * \since Version 1.24
+   */
+  ORT_API2_STATUS(GetOrtFenceForGraphicsInterop, _In_ OrtSession* session, _In_ const struct GraphicsInteropParams* graphicsInteropParams, _In_ struct FenceInteropParams* fenceInteropParams, _Outptr_ OrtFence** ortFence);
```

Comment:
Who owns `ortFence` and how/when should it be released?

Comments seem to be out of date.

Why do fenceInteropParams need to be non-const?